src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -311,132 +197,82 @@ public sealed class MempoolBlobTxValidator : ITxValidator
     public static readonly MempoolBlobTxValidator Instance = new();
     private MempoolBlobTxValidator() { }
 
-    public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec, [NotNullWhen(false)] out string? error)
+    public ValidationResult IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec)
     {
-        error = null;
-
-        if (transaction.NetworkWrapper is not ShardBlobNetworkWrapper wrapper)
-        {
-            return true;
-        }
-
-        int blobCount = transaction.BlobVersionedHashes.Length;
-        if (wrapper.Blobs.Length != blobCount)
-        {
-            error = TxErrorMessages.InvalidBlobData;
-            return false;
-        }
-
-        if (wrapper.Commitments.Length != blobCount)
-        {
-            error = TxErrorMessages.InvalidBlobData;
-            return false;
-        }
-
-        if (wrapper.Proofs.Length != blobCount)
-        {
-            error = TxErrorMessages.InvalidBlobData;
-            return false;
-        }
-
-        for (int i = 0; i < blobCount; i++)
+        int blobCount = transaction.BlobVersionedHashes!.Length;
+        return transaction.NetworkWrapper is not ShardBlobNetworkWrapper wrapper ? ValidationResult.Success
+            : wrapper.Blobs.Length != blobCount ? TxErrorMessages.InvalidBlobData
+            : wrapper.Commitments.Length != blobCount ? TxErrorMessages.InvalidBlobData
+            : wrapper.Proofs.Length != blobCount ? TxErrorMessages.InvalidBlobData
+            : ValidateBlobs();
```

Comment:
For consistency I would go for a switch expression like in `BlobFieldsTxValidator`